includes/class-files-integrity.php

```diff
@@ -0,0 +1,187 @@
+	 *
+	 * @return void
+	 */
+	static function run_files_integrity_check() {
+
+		$checksums = Files_Integrity::call_checksum_api();
+
+		$files = Files_Integrity::parse_checksum_results( $checksums );
+
+		Files_Integrity::create_the_response( $files );
+
+	}
+
+	/**
+	* Calls the WordPress API on the checksums endpoint
+	*
+	* @uses get_bloginfo()
+	* @uses get_locale()
+	* @uses ABSPATH
+	* @uses wp_remote_get()
+	* @uses get_bloginfo()
+	* @uses strpos()
+	* @uses unset()
+	*
+	* @return array
+	*/
+	static function call_checksum_api() {
+		// Setup variables.
+		$wpversion = get_bloginfo( 'version' );
+		$wplocale  = get_locale();
+
+		// Setup API Call.
+		$checksumapi = wp_remote_get( 'https://api.wordpress.org/core/checksums/1.0/?version=' . $wpversion . '&locale=' . $wplocale );
+
+		// Encode the API response body.
+		$checksumapibody = json_decode( wp_remote_retrieve_body( $checksumapi ), true );
+
+		// Remove the wp-content/ files from checking
+		foreach ( $checksumapibody['checksums'] as $file => $checksum ) {
+			if ( false !== strpos( $file, 'wp-content/' ) ) {
+				unset( $checksumapibody['checksums'][ $file ] );
+			}
+		}
+
+		return $checksumapibody;
+	}
+
+	/**
+	* Parses the results from the WordPress API call
+	*
+	* @uses file_exists()
+	* @uses md5_file()
+	* @uses ABSPATH
+	*
+	* @param array $checksums
+	*
+	* @return array
+	*/
+	static function parse_checksum_results( $checksums ) {
+		$filepath = ABSPATH;
+		$files    = array();
+		// Parse the results.
+		foreach ( $checksums['checksums'] as $file => $checksum ) {
+			// Check the files.
+			if ( file_exists( $filepath . $file ) && md5_file( $filepath . $file ) !== $checksum ) {
+				$reason = esc_html__( 'Content changed', 'health-check' ) . ' <a href="#health-check-diff" data-file="' . $file . '">' . esc_html__( '(View Diff)', 'health-check' ) . '</a>';
+				array_push( $files, array( $file, $reason ) );
+			} elseif ( ! file_exists( $filepath . $file ) ) {
+				$reason = esc_html__( 'File not found', 'health-check' );
+				array_push( $files, array( $file, $reason ) );
+			}
+		}
+		return $files;
+	}
+
+	/**
+	* Generates the response
+	*
+	* @uses wp_send_json_success()
+	* @uses wp_die()
+	* @uses ABSPATH
+	*
+	* @param null|array $files
+	*
+	* @return array
+	*/
+	static function create_the_response( $files ) {
+		$filepath = ABSPATH;
+		$output   = '';
+
+		if ( empty( $files ) ) {
+			$output .= '<div class="notice notice-success inline"><p>';
+			$output .= esc_html__( 'All files passed the check. Everything seems to be ok!', 'health-check' );
+			$output .= '</p></div>';
+		} else {
+			$output .= '<div class="notice notice-error inline"><p>';
+			$output .= __( 'It appears that some files have been tampered with. Please either update WordPress or manually replace the files you see on the list and run the <code>File Integrity</code> check again.', 'health-check' );
```

Comment:
Let's re-word this to make it a little bit less frightening. I'm thinking `It appears that some files may have been modified.`, thoughts?

Reply:
You've got a point. I went full explanatory and to be honest if we're talking about people that won't understand what has changed in those files, they might update something and break things even more. So I'll change the whole notice into just `It appears that some files may have been modified.`.

Since sometimes less is more a simple informative notice to make them ask in the forums or their dev is better ^_^.